app/services/embedded_content_finder_service.rb

```diff
@@ -8,20 +8,20 @@ class EmbeddedContentFinderService
   def fetch_linked_content_ids(details, locale)
     content_references = details.values.map { |value|
       find_content_references(value)
-    }.flatten.compact.uniq
+    }.flatten.compact
 
-    check_all_references_exist(content_references, locale)
+    check_all_references_exist(content_references.uniq, locale)
     content_references.map(&:content_id)
   end
 
   def find_content_references(value)
     case value
     when Array
-      value.map { |item| find_content_references(item) }.flatten
+      value.map { |item| find_content_references(item) }.flatten.uniq
```

Comment:
is `find_content_references` basically recursively called until the `value` becomes a string? what's the impact of adding `uniq` on the end for arrays and hashes?

Reply:
So, this is mainly for multipart content, when there's a html and govspeak part. We don't want to class these as two instances